notebooks/12-xarray-advanced.md

```diff
@@ -0,0 +1,598 @@
+
++++
+
+Adding new variables to the data set is very similar to Pandas/GeoPandas:
+
+```{code-cell} ipython3
+ds["precipitation_m"] = ds["precipitation"]/1000.
+```
+
+```{code-cell} ipython3
+ds
+```
+
+### Selecting `DataSet` data
+
++++
+
+Each of the data variables can be accessed as a single `xarray.DataArray` similar to selecting dictionaries or DataFrames:
+
+```{code-cell} ipython3
+type(ds["precipitation"]), type(ds["temperature"])
+```
+
+One can select multiple variables at the same time as well by passing a list of variable names:
+
+```{code-cell} ipython3
+ds[["temperature", "precipitation"]]
+```
+
+Or the other way around, use the `drop_vars` to drop variables from the data set:
+
+```{code-cell} ipython3
+ds.drop_vars("temperature")
+```
+
+<div class="alert alert-info" style="font-size:120%">
+
+**NOTE**: <br>
+
+Selecting a single variable using `[]` results into a `xarray.DataArray`, selecting multiple variables using a list `[[..., ...]]` results into a `xarray.DataSet`. Using `drop_vars` always returns a `xarray.DataSet`.
+
+</div>
+
++++
+
+The selection with `sel` works as well with `xarray.DataSet`, selecting the data _for all variables in the DataSet_ and returning a DataSet:
+
+```{code-cell} ipython3
+ds.sel(year=2016)
+```
+
+The inverse of the `sel` method is the `drop_sel` which returns a DataSet with the enlisted indices removed:
+
+```{code-cell} ipython3
+ds.drop_sel(year=[2016])
+```
+
+### DataSet plotting
+
++++
+
+Plotting for data set level is rather limited. A typical use case that is supported to compare two data variables are scatter plots:
+
+```{code-cell} ipython3
+ds.plot.scatter("temperature", "precipitation", s=1, alpha=0.1)
+```
+
+```{code-cell} ipython3
+ds.plot.scatter("temperature", "precipitation", s=1, alpha=0.1, col="year")  # try also hue instead of col; requires hue_style="discrete"
+```
+
+### DataSet reductions
+
++++
+
+Datasets support arithmetic operations by automatically looping over all data variables and supports most of the same methods found on `xarray.DataArray`:
+
+```{code-cell} ipython3
+ds.mean()
+```
+
+```{code-cell} ipython3
+ds.max(dim=["x", "y"])
+```
+
+__Note__ Using the names of the data variables (which is actually element-wise operations with DataArrays) makes a calculation very self-describing, e.g.
+
+```{code-cell} ipython3
+(ds["temperature"] * ds["precipitation"]).sel(year=2016).plot.imshow()
+```
+
+### Let's practice
+
++++
+
+For the next set of exercises, we introduce the [ERA5-Land monthly averaged data](https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-land-monthly-means?tab=overview). 
+
+> ERA5-Land is a reanalysis dataset providing a consistent view of the evolution of land variables over several decades. Reanalysis combines model data with observations from across the world into a globally complete and consistent dataset using the laws of physics. 
```

Comment:
Maybe add that is it provided by the ECMWF (European Centre for Medium-Range Weather Forecasts).

Reply:
ok, done